R/install-github.r

```diff
@@ -53,14 +55,48 @@
 install_github <- function(repo, username = NULL,
                            ref = "master", subdir = NULL,
                            auth_token = github_pat(),
-                           host = "api.github.com", ...) {
+                           host = "api.github.com",
+                           skip_same = TRUE, ...) {
 
   remotes <- lapply(repo, github_remote, username = username, ref = ref,
     subdir = subdir, auth_token = auth_token, host = host)
 
+  if (isTRUE(skip_same)) {
+    remotes <- Filter(different_sha, remotes)
+  }
+
   install_remotes(remotes, ...)
 }
 
+different_sha <- function(x) {
+
+  github_sha <- github_commit(x$username, x$repo, x$ref)$sha
+
+  # download DESCRIPTION from GitHub
+  github_description <- github_DESCRIPTION(x$username, x$repo, github_sha)
+
+  # retrieve DESCRIPTION of installed package
+  # We need to use the package name from the GitHub DESCRIPTION in case the
+  # GitHub repository name differs from that of the package name.
+  local_description <-
+    suppressWarnings(packageDescription(github_description$Package))
+
+  # Package not installed
+  not_installed <- is.na(local_description)
```

Comment:
Could we move this up and use `is_installed()`? I think that would eliminate the need for `suppressWarnings()`